.travis.yml

```diff
@@ -1,31 +1,63 @@
 language: ruby
+
 sudo: false
 dist: trusty
+
 rvm:
   - 2.3.0
+
+# Allow Emacs snapshot builds to fail and don't wait for it
 matrix:
   fast_finish: true
   allow_failures:
     - env: EMACS_VERSION=snapshot
+
+# Cache stable Emacs binaries and packages
 cache:
-  - bundler
-  - apt: true
-  - directories:
+  bundler: true
+  apt: true
+  timeout: 691200
+  directories:
       - "$HOME/emacs"
+      - "$HOME/.emacs.d"
+      - vendor/bundle
+
+git:
+  depth: 1
+
 env:
   matrix:
+    - EMACS_VERSION=24.3
+    - EMACS_VERSION=24.4
     - EMACS_VERSION=24.5
-    - EMACS_VERSION=25.1-rc2
```

Comment:
I agree that there's little point to test for `25.1-rc2` now that `25.1` is out.  However, I left the test to make sure that the `emacs-travis.mk` makefile was still able to to build release candidates.  Please restore this one.